frontend/src/services/GooglePlacesService.ts

```diff
@@ -22,13 +27,40 @@ class GooglePlacesService {
             input: dataURI,
             inputtype: "textquery",
             fields: "place_id",
-            key: process.env.GOOGLE_PLACES_API_KEY,
+            key: process.env.REACT_APP_GOOGLE_PLACES_API_KEY,
           },
         }
       )
       .then((response: AxiosResponse) => {
-        const resultPlaceID: string = response.data.candidates?.[0].place_id;
-        return Promise.resolve(resultPlaceID);
+        if (response.data.candidates.length === 1) {
+          return Promise.resolve(response.data.candidates?.[0].place_id);
```

Comment:
Does `response.data.candidates[0].place_id` (without the optional chaining) not work here? I'm assuming `response.data.candidates` is defined at this point but I could be wrong!

Reply:
Sorry 😅  what are you asking about in particular? I'm checking if the length here is 1 because that means there weren't any other restaurants found with similar names (either because it's a franchise or just multiple restaurants with that name)